src/Approximations/underapproximate.jl

```diff
@@ -0,0 +1,29 @@
+"""
+    underapproximate(X::LazySet{N}, dirs::AbstractDirections;
+                    [apply_convex_hull]::Bool=false) where {N<:Real}
+
+Compute the underapproximation of a convex set by sampling support vectors.
+
+### Input
+
+- `X`                 -- set
+- `dirs`              -- directions
+- `apply_convex_hull` -- (optional, default: `false`) if `true`, post-process
+                         the support vectors with a convex hull operation
+### Output
+
+The `VPolytope` obtained by taking the convex hull of the support vectors of `X`
+along the directions determined by `dirs`.
+"""
+function underapproximate(X::LazySet{N}, dirs::AbstractDirections;
+                          apply_convex_hull::Bool=false) where {N<:Real}
+    @assert LazySets.dim(X) == LazySets.dim(dirs)
+    vinner = Vector{Vector{N}}(undef, length(dirs)) # TODO could be generalized to eltype(dirs)
```

Comment:
The reason that `eltype(dirs)` doesn't currently work here is that `eltype` of directions is other than `Vector` in general, while `VPolytope`'s `vertices` fields is a `vertices::Vector{Vector{N}}`. That should be eventually generalized.

Reply:
Not sure about the TODO. The result type of σ also depends on X.